### ISSUE TITLE
ListBox does not work with readers

### ISSUE THREAD

**Issue by ag-petrov** [telerik/kendo-ui-core#6371]
### Bug report

When the BindTo method is used as in [this ](https://demos.telerik.com/aspnet-mvc/listbox)demo the screen readers will fail to announce the focused item. More information on the matter can be found in ticket - 1514198

### Reproduction of the problem
Run the basic usage demo linked above and navigate inside the listview using a screen reader

### Current behavior
Focused item is not announced

### Expected/desired behavior
Focused item should be announced

**Comment by dmanova** [comment]
It seems the screen readers announce the focused item if it has "id" attribute in the corresponding < li > tag.  Indeed in the following cases, the "id" is not set:
- For items in the right listbox
- When Draggable method is used
- When BindTo method is used

**Comment by dmanova** [comment]
Here is a workaround that set up id attributes for the ListBox items with JavaScript function, that may be attached to the DataBound event:

```
<script>        
function onDataBound(e) {
           $("div.k-list-scroller.k-selectable > ul > li").each(function () {
                var randomId = Math.floor(Math.random() * 10000001);
                $(this).attr("id", randomId);
            });   
        }
</script>
```

**Comment by Juveniel** [comment]
The two described issues by @dmanova  for connected ListBox and draggable are fixed.

In regards to the BindTo() method issue - when binding the ListBox to a simple strings collection,  the DataSource is not capable of generating unique identifiers for the items while processing the data. In order for the ListBox to generate `uid'` for the items the component should be bound to a list of objects e.g:

```cshtml
ViewBag.Attendees = new List<SelectListItem>
{
    new SelectListItem(){ Value = "1", Text = "Steven White" },
    new SelectListItem(){ Value = "2", Text = "Nancy King" },
    new SelectListItem(){ Value = "3", Text = "Nancy Davolio" },
    new SelectListItem(){ Value = "4", Text = "Michael Leverling" },
    new SelectListItem(){ Value = "5", Text = "Andrew Callahan" },
    new SelectListItem(){ Value = "6", Text = "Michael Suyama" }
};
```

With the above configuration, the items will have `data-uid` and `id` attributes assigned which will resolve the accessibility issues in this particular scenario. To better reflect this, we will also updated the Basic Usage Demos for MVC and Core.
